### PR TITLE
[WIP] Feature-Gate affinity in annotations

### PR DESCRIPTION
**Pull request by rrati**
WIP to enable affinity in annotations

### CODE REVIEW

**Comment by rrati** [comment]
@timothysc

**Comment by davidopp** [comment]
cc/ @wojtek-t

**Comment by timothysc** [comment]
I'm going to take this PR and clean it up, b/c @rrati will be on vacation for several weeks.

**Comment by davidopp** [comment]
IIUC @timothysc said he is going to take this over.

**Comment by davidopp** [comment]
For reference, the PRs that removed annotation-based affinity (which this PR is supposed to add back under feature gate) are #37299 #39478

An example of the way to do feature-gating is https://github.com/kubernetes/kubernetes/pull/30468
Affinity can be a single "feature" (don't need to split node/pod)

**Comment by timothysc** [comment]
I'll create a new PR from this original one today.  There is not much more work on this.

File: staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate.go [hunk 1]
```diff
@@ -107,6 +107,8 @@ type FeatureGate interface {
 	// owner: @pweil-
 	// alpha: v1.5
 	ExperimentalHostUserNamespaceDefaulting() bool
+
+	AnninityInAnnotations() bool
```
Comment (ncdc) [inline staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate.go]: Anninity 😄

Comment (timothysc) [inline staging/src/k8s.io/apiserver/pkg/util/feature/feature_gate.go]: Yeah I already fixed that ;-)

**Comment by timothysc** [comment]
Closed in preference of #41617

**Pull request closed without merging**